go/tasks/pluginmachinery/flytek8s/pod_helper.go

```diff
@@ -96,27 +96,54 @@ func BuildIdentityPod() *v1.Pod {
 //          resources requested is beyond the capability of the system. for this we will rely on configuration
 //          and hence input gates. We should not allow bad requests that request for large number of resource through.
 //          In the case it makes through, we will fail after timeout
-func DemystifyPending(status v1.PodStatus) (pluginsCore.PhaseInfo, error) {
+func DemystifyPending(pod *v1.Pod) (pluginsCore.PhaseInfo, error) {
 	// Search over the difference conditions in the status object.  Note that the 'Pending' this function is
 	// demystifying is the 'phase' of the pod status. This is different than the PodReady condition type also used below
+	status := pod.Status
+	timeout := config.GetK8sPluginConfig().MaxSystemLevelTimeout
 	for _, c := range status.Conditions {
 		switch c.Type {
 		case v1.PodScheduled:
 			if c.Status == v1.ConditionFalse {
+				// If the pod is interruptible and is waiting to be scheduled for an extended amount of time,  it is possible there are
```

Comment:
ConditionFalse _I think_ means this condition is no longer the "current" but it'll never be removed from the list of conditions. I do not think we should apply the timeout in this case...

Reply:
Yep took a look at the docs and it appears it means that the condition is not applicable. Removed.